clang/runtime/dpct-rt/include/dnnl_utils.hpp.inc

```diff
@@ -2147,6 +2277,44 @@ public:
 };
 // DPCT_LABEL_END
 
+// DPCT_LABEL_BEGIN|dropout_desc_def|dpct::dnnl
+// DPCT_DEPENDENCY_EMPTY
+// DPCT_CODE
+inline
+void dropout_desc::restore(engine_ext &engine, float p, void *state,
+                           size_t state_size, unsigned long long seed) {
+  _p = p;
+  _seed = seed;
+  if (state) {
+    _state = state;
+    sycl::queue *q = engine.get_queue();
+    _rng_engine = new oneapi::mkl::rng::philox4x32x10(
+        oneapi::mkl::rng::load_state<oneapi::mkl::rng::philox4x32x10>(*q,
+                                                                      (std::uint8_t *)state));
+    _distr = new oneapi::mkl::rng::bernoulli<
+        std::int32_t, oneapi::mkl::rng::bernoulli_method::icdf>(p);
+  }
+}
+
+inline
+void dropout_desc::set(engine_ext &engine, float p, void *state,
+                       size_t state_size, unsigned long long seed) {
+  _p = p;
+  _seed = seed;
+  if (state) {
+    _state = state;
+    sycl::queue *q = engine.get_queue();
+    _rng_engine = new oneapi::mkl::rng::philox4x32x10(*q, seed);
+    _distr = new oneapi::mkl::rng::bernoulli<
+        std::int32_t, oneapi::mkl::rng::bernoulli_method::icdf>(p);
+    if (state_size < oneapi::mkl::rng::get_state_size(*_rng_engine)) {
```

Comment:
Maybe we can store the state_size as a `static const` variable in `dropout_desc`?